general/macros.hpc

```diff
@@ -244,8 +254,18 @@ set     ModulesVersion      "%{version}"\
 EOF\
 %nil
 
-#
-%hpc_make_modules_dir mkdir -p %{buildroot}%{hpc_module_dep_base}
+# make the modules dir
+# if we are called inside python_expand, possibly provide modules for extra flavors
+%hpc_make_modules_dir \
+mkdir -p %{buildroot}%{hpc_module_dep_base} \
+if [ -n "%{?$python_provides}"]; then \
```

Comment:
Hrm, this is ugly - and everybody who will look at this after me will stumble over this as well. 
Especially, since this is not a macro  specific to python-singlespec.
Isn't there anything on the macro level that can be checked to detect if the expansion happens inside a %python_expand and only expand the shell code then? This together with a comment would keep the macros more maintainable.

Reply:
The iteration is executed if and only if `%{?$python_provides}` expands to more than the null string, i.e. only within python_expand and only for the flavor where this is defined. Non-singlespec python packages obviously don't care about extra flavor providers and need to define their tags and extra files separately.